flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/files/StaticFileServerHandler.java

```diff
@@ -304,8 +314,15 @@ private void respondAsLeader(ChannelHandlerContext ctx, HttpRequest request, Str
 		ctx.write(response);
 
 		// write the content.
-		ctx.write(new DefaultFileRegion(raf.getChannel(), 0, fileLength), ctx.newProgressivePromise());
-		ChannelFuture lastContentFuture = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+		ChannelFuture lastContentFuture;
+		if (ctx.pipeline().get(SslHandler.class) == null) {
+			ctx.write(new DefaultFileRegion(raf.getChannel(), 0, fileLength), ctx.newProgressivePromise());
+			lastContentFuture = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
+		} else {
+			lastContentFuture = ctx.writeAndFlush(new HttpChunkedInput(new ChunkedFile(raf, 0, fileLength, 8192)),
+				ctx.newProgressivePromise());
+			// HttpChunkedInput will write the end marker (LastHttpContent) for us.
```

Comment:
This is based on the Netty example code - https://netty.io/4.0/xref/io/netty/example/http/file/HttpStaticFileServerHandler.html

The initial flink code somehow did not include these checks. Cannot do zero copy file transfer with SSL enabled and the existing stream handling code hangs without any data transfer. Hence had to introduce this.